backend/mem/location_test.go

```diff
@@ -0,0 +1,186 @@
+	"regexp"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/c2fo/vfs/v4"
+	"github.com/c2fo/vfs/v4/utils"
+)
+
+/**********************************
+ ************TESTS*****************
+ **********************************/
+
+type osLocationTest struct {
+	suite.Suite
+	testFile   vfs.File
+	fileSystem *FileSystem
+}
+
+
+func (s *osLocationTest) SetupSuite() {
+	setupTestFiles()
+}
+
+func (s *osLocationTest) TearDownSuite() {
+	teardownTestFiles()
+}
+/*
+func (s *osLocationTest) SetupTest() {
+	fs := &FileSystem{}
+	file, err := fs.NewFile("", "test_files/test.txt")
+
+	if err != nil {
+		s.Fail("No file was opened")
+	}
+
+	s.testFile = file
+	s.fileSystem = fs
+}
+*/
+func (s *osLocationTest) TestList() {
+
+	expected := []string{"empty.txt", "prefix-file.txt", "test.txt"}
+	actual, _ := s.testFile.Location().List()
+	s.Equal(expected, actual)
+}
+
+func (s *osLocationTest) TestList_NonExistentDirectory() {
+	location, err := s.testFile.Location().NewLocation("not/a/directory/")
+	s.Nil(err, "error isn't expected")
+
+	exists, err := location.Exists()
+	s.Nil(err, "error isn't expected")
+	s.False(exists, "location should return false for Exists")
+
+	contents, err := location.List()
+	s.Nil(err, "error isn't expected")
+	s.Equal(0, len(contents), "List should return empty slice for non-existent directory")
+
+	prefixContents, err := location.ListByPrefix("anything")
+	s.Nil(err, "error isn't expected")
+	s.Equal(0, len(prefixContents), "ListByPrefix should return empty slice for non-existent directory")
+
+	regex, _ := regexp.Compile("[-]+")
+	regexContents, err := location.ListByRegex(regex)
+	s.Nil(err, "error isn't expected")
+	s.Equal(0, len(regexContents), "ListByRegex should return empty slice for non-existent directory")
+}
+
+func (s *osLocationTest) TestListByPrefix() {
+	expected := []string{"prefix-file.txt"}
+	actual, _ := s.testFile.Location().ListByPrefix("prefix")
+	s.Equal(expected, actual)
+
+	_, err := s.testFile.Location().ListByPrefix("bad/prefix")
+	s.EqualError(err, utils.BadFilePrefix, "got expected error")
+}
+
+func (s *osLocationTest) TestListByRegex() {
+	expected := []string{"prefix-file.txt"}
+	regex, _ := regexp.Compile("[-]+")
+	actual, _ := s.testFile.Location().ListByRegex(regex)
+	s.Equal(expected, actual)
+}
+
+func (s *osLocationTest) TestExists() {
+	otherFile, _ := s.fileSystem.NewFile("", "foo/foo.txt")
+	s.True(s.testFile.Location().Exists())
+	s.False(otherFile.Location().Exists())
+}
+
+func (s *osLocationTest) TestNewLocation() {
+	otherFile, _ := s.fileSystem.NewFile("", "/foo/foo.txt")
+	fileLocation := otherFile.Location()
+	subDir, _ := fileLocation.NewLocation("other/")
+	s.Equal("/foo/other/", subDir.Path())
+
+	relDir, _ := subDir.NewLocation("../../bar/")
+	s.Equal("/bar/", relDir.Path(), "relative dot path works")
+}
+
+func (s *osLocationTest) TestNewFile() {
+	//loc, err := s.fileSystem.NewLocation("", "/foo/bar/baz/")
```

Comment:
Same comment as above here, should this test be removed, or will it be fixed in a later commit?